shell/src/main/java/alluxio/cli/fsadmin/quorum/QuorumInfoCommand.java

```diff
@@ -0,0 +1,124 @@
+import alluxio.grpc.JournalDomain;
+import alluxio.grpc.QuorumServerInfo;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Command for querying journal quorum information.
+ */
+public class QuorumInfoCommand extends AbstractFsAdminCommand {
+  public static final String DOMAIN_OPTION_NAME = "domain";
+
+  public static final String OUTPUT_HEADER_DOMAIN = "Journal domain\t: %s";
+  public static final String OUTPUT_HEADER_QUORUM_SIZE = "Quorum size\t: %d";
+  public static final String OUTPUT_SERVER_INFO = "%s\t %s:%s";
+
+  /**
+   * @param context fsadmin command context
+   * @param alluxioConf Alluxio configuration
+   */
+  public QuorumInfoCommand(Context context, AlluxioConfiguration alluxioConf) {
+    super(context);
+  }
+
+  /**
+   * @return command's description
+   */
+  @VisibleForTesting
+  public static String description() {
+    return "Shows quorum information for embedded journal.";
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    // Validate domain option is correct.
+    String domainVal = cl.getOptionValue(DOMAIN_OPTION_NAME);
+    try {
+      JournalDomain.valueOf(domainVal);
+    } catch (IllegalArgumentException e) {
+      throw new InvalidArgumentException(ExceptionMessage.INVALID_OPTION_VALUE
+          .getMessage(DOMAIN_OPTION_NAME, Arrays.toString(JournalDomain.values())));
+    }
+
+    JournalMasterClient jmClient = mMasterJournalMasterClient;
+
+    if (cl.hasOption(DOMAIN_OPTION_NAME)) {
+      JournalDomain domain = JournalDomain.valueOf(domainVal);
+      if (domain == JournalDomain.JOB_MASTER) {
+        jmClient = mJobMasterJournalMasterClient;
+      }
+    }
+
+    GetQuorumInfoPResponse quorumInfo = jmClient.getQuorumInfo();
+    mPrintStream.println(String.format(OUTPUT_HEADER_DOMAIN, quorumInfo.getDomain()));
+    mPrintStream
+        .println(String.format(OUTPUT_HEADER_QUORUM_SIZE, quorumInfo.getServerInfoList().size()));
+    mPrintStream.println("STATE\t\tSERVER ADDRESS");
+    for (QuorumServerInfo serverState : quorumInfo.getServerInfoList()) {
+      String serverStateStr = String.format(OUTPUT_SERVER_INFO, serverState.getServerState(),
+          serverState.getServerAddress().getHost(), serverState.getServerAddress().getRpcPort());
+      mPrintStream.println(serverStateStr);
+    }
+
+    return 0;
+  }
+
+  @Override
+  public String getCommandName() {
+    return "info";
+  }
+
+  @Override
+  public String getUsage() {
+    return String.format("%s -%s <%s>%n", getCommandName(), DOMAIN_OPTION_NAME,
+        Arrays.toString(JournalDomain.values()));
+  }
+
+  @Override
+  public String getDescription() {
+    return description();
+  }
+
+  @Override
+  public void validateArgs(CommandLine cl) throws InvalidArgumentException {
+    if (cl.getOptions().length != 1) {
```

Comment:
Could we make `master` domain the default? Usually users are not that interest in job_master domain compared to master.

Reply:
I thought about that but couldn't find a nice way to plug optional arguments into our command subsystem and didn't want to complicate the code. I feel like it's lay out arguments explicitly for this particular command as it's for advanced use. I don't have a strong preference though.